transcriptomics/transcriptomics_matrix_generator.py

```diff
@@ -91,14 +100,23 @@ def split_into_groups(self, biosamples, num_groups, max_size=None):
         return groups
 
     def generate_counts_matrix(self, differential_expr_percentage=10, differential_factor=2.5, dispersion=0.2, outlier_percentage=5, outlier_factor=10):
+        if not self.gene_names:
+            raise ValueError("No gene names available for count matrix generation.")
+
+        unique_gene_names = list(filter(None, set(self.gene_names)))
+        if not unique_gene_names:
+            raise ValueError("No valid gene names available after filtering duplicates and empty entries.")
```

Comment:
I don't think this is needed, since the values are already deduplicated when the `self.gene_names` property is set.